nodebuilder/gateway/module.go

```diff
@@ -0,0 +1,60 @@
+package gateway
+
+import (
+	"context"
+
+	"go.uber.org/fx"
+
+	"github.com/celestiaorg/celestia-node/api/gateway"
+	headerServ "github.com/celestiaorg/celestia-node/nodebuilder/header"
+	"github.com/celestiaorg/celestia-node/nodebuilder/node"
+	shareServ "github.com/celestiaorg/celestia-node/nodebuilder/share"
+	stateServ "github.com/celestiaorg/celestia-node/nodebuilder/state"
+)
+
+func ConstructModule(tp node.Type, cfg *Config) fx.Option {
+	// sanitize config values before constructing module
+	cfgErr := cfg.Validate()
+
+	if !cfg.Enabled {
+		return fx.Options()
+	}
+
+	baseComponents := fx.Options(
+		fx.Supply(cfg),
```

Comment:
We need to clean up Config usage pattern in our module configs. Somewhere we use ptr, and somewhere a value. Let's use ptr everywhere.